Source/WebKit/Platform/IPC/MessageSender.h

```diff
@@ -70,6 +70,10 @@ class MessageSender {
     template<typename T, typename C, typename U, typename V> inline AsyncReplyID sendWithAsyncReply(T&& message, C&& completionHandler, ObjectIdentifierGeneric<U, V> destinationID);
     template<typename T, typename C, typename U, typename V> inline AsyncReplyID sendWithAsyncReply(T&& message, C&& completionHandler, ObjectIdentifierGeneric<U, V> destinationID, OptionSet<SendOption>);
 
+    template<typename T> Ref<typename T::Promise> inline sendWithPromisedReply(T&& message);
+    template<typename T> Ref<typename T::Promise> inline sendWithPromisedReply(T&& message, uint64_t destinationID);
+    template<typename T> Ref<typename T::Promise> inline sendWithPromisedReply(T&& message, uint64_t destinationID, OptionSet<SendOption>);
+
```

Comment:
Can we use optional parameters like `OptionSet<SendOption> = { }`?

Reply:
I copied the style and setup of the sendWithAsyncReply. 

I’m not sure why it wasn’t define to use optional parameters there